iree/compiler/Dialect/Flow/Transforms/ConvertToFlowTensorOps.cpp

```diff
@@ -292,12 +110,27 @@ struct ConvertToFlowTensorOpsPass
     FuncOp funcOp = getOperation();
     MLIRContext *context = funcOp->getContext();
     context->allowUnregisteredDialects(true);
+
+    // Run partial conversion from tensor -> flow.tensor ops.
+    OwningRewritePatternList conversionPatterns(&getContext());
+    TypeConverter typeConverter;
+    typeConverter.addConversion([](Type t) { return t; });
+    ConversionTarget target(*context);
+    setupTensorToFlowLegality(context, target, typeConverter);
+    populateTensorToFlowPatterns(&getContext(), conversionPatterns,
+                                 typeConverter);
+    if (failed(applyPartialConversion(getOperation(), target,
```

Comment:
This pass flow is strange. It is first doing dialect conversion , and then doing pattern rewrites. Made me go back and see the changes in `ConvertTensorToFlow` and see that these are dialect conversions. Maybe making the patterns `OpRewritePatterns` and can avoid doing all the dialect conversion boiler plate, and you can just add these patterns to run "before dispatch region" formation case. Then you will have a single `applyPatternsAndFoldGreedily`